hirefire/procs/__init__.py

```diff
@@ -65,19 +66,23 @@ def native_dump_procs(procs):
     a list of dictionaries in the form expected by HireFire,
     ready to be encoded into JSON.
     """
-    data = []
     cache = {}
-    for name, proc in procs.items():
+    
+    def _run(args):
+        name, proc = args
         try:
             quantity = proc.quantity(cache=cache)
         except TypeError:
             quantity = proc.quantity()
-
-        data.append({
+            return {
             'name': name,
             'quantity': quantity or 0,
-        })
-    return data
+        }
+        
+    with ThreadPoolExecutor() as executor:
+        # Execute all procs in parallel to avoid blocking IO
+        # especially celery which needs to open a transport to AMQP.
+        return list(executor.map(_run, procs.items()))
```

Comment:
Is there any downside to this? Are the app and connection used by Celery thread-safe? We use the same app that celery does to get the connection, etc. I don't want to end up flooding the RabbitMQ server with connections, and I'm just trying to think of other possible downsides. I'm always slow to use threads, because I don't use them very often and I don't have a lot of practice with thinking through possible issues with them.

Reply:
That is a good point. I recently had an issue with connection stacking regarding the DB even if the DB was not used, see https://github.com/KristianOellegaard/django-health-check/pull/182

I checked both [amqp](http://amqp.readthedocs.io/en/latest/_modules/Queue.html?highlight=thread) and librabbitmq which uses rabbitmq-c which seem to be thread safe.

The DB issue only happens in configuration edge cases, but I am happy to add the bit that ensures they are closed if you want me to.

Reply:
Does that mean that you have not yet used this branch in production? How will the affect other Procs, for queues other than Celery?

Reply:
The only other Proc we use uses Redis. I can not vouch that this will not have any side effects on other third party packages.
We are currently not using it on production, I will setup a staging environment tho and stress test the endpoint.

Can you assign the PR to me, that way I won't for get. I am going on vacation on Friday. I don't know if I can manage before I leave.

Reply:
@codingjoe : Have you been able to test it in a staging or production environment?